pyrealm/pmodel/subdaily.py

```diff
@@ -322,6 +323,14 @@ def __init__(
             1 / calc_ftemp_arrh(tk_acclim, self.env.pmodel_const.subdaily_jmax25_ha)
         )
 
+        """Instantaneous optimal :math:`x_{i}`, :math:`V_{cmax}` and :math:`J_{max}`"""
+        if init_realised is not None:
+            self.init_xi_real, self.init_vcmax_real, self.init_jmax_real = init_realised
```

Comment:
This needs to validate the provided values - mostly just that they are the correct shape.  I'm not sure they need to be stored as class attributes.

Reply:
Do you mean something like
```
if init_realised is not None:
    assert(init_realised[0].shape() ==self.xi_real.shape())
    assert(init_realised[1].shape() == self.vcmax25_real.shape())
    assert(init_realised[2].shape() == self.jmax25_real.shape())
    init_xi_real, init_vcmax_real, init_jmax_real = init_realised
```

Reply:
Something like that yes - but with an Exception not `assert` 😄

Reply:
OK. Why would you use an exception here?

Reply:
IIUC `assert` shouldn't be used in production code, because it is sensitive to the way in which `python` is invoked. If the code is run using `python -o` or `python -oo` then those assert statements are discarded and any bad data would sail through.

https://docs.python.org/3/using/cmdline.html#cmdoption-O

Reply:
I changed the shape check to

```
        if init_realised is not None:
            if not (
                (init_realised[0].shape() == self.xi_real[0].shape())
                and (init_realised[1].shape() == self.vcmax25_real[0].shape())
                and (init_realised[2].shape() == self.jmax25_real[0].shape())
            ):
                raise Exception("`init_realised` has wrong shape in Subdaily PModel")
            else:
                init_xi_real, init_vcmax_real, init_jmax_real = init_realised
        else:
            init_xi_real, init_vcmax_real, init_jmax_real = [None, None, None]
```

Does that make sense?